libtrip-notification/src/test/java/com/mapbox/navigation/trip/notification/MapboxTripNotificationTest.kt

```diff
@@ -124,4 +136,145 @@ class MapboxTripNotificationTest {
 
         verify(exactly = 1) { NavigationNotificationProvider.buildNotification(any()) }
     }
+
+    @Test
+    fun whenUpdateNotificationCalledThenBannerInstructionsPrimaryTextInteracted() {
+        val routeProgress = mockk<RouteProgress>(relaxed = true)
+        val primaryText = { "Primary Text" }
+        val bannerText = mockBannerText(routeProgress, primaryText)
+        mockUpdateNotificationAndroidInteractions()
+
+        notification.updateNotification(routeProgress)
+
+        verify(exactly = 1) { bannerText.text() }
+        verify(exactly = 2) { anyConstructed<RemoteViews>().setTextViewText(any(), primaryText()) }
+    }
+
+    @Test
+    fun whenUpdateNotificationCalledTwiceWithSameDataThenRemoteViewAreNotUpdatedTwice() {
+        val routeProgress = mockk<RouteProgress>(relaxed = true)
+        val primaryText = { "Primary Text" }
+        val bannerText = mockBannerText(routeProgress, primaryText)
+        mockUpdateNotificationAndroidInteractions()
+
+        notification.updateNotification(routeProgress)
+
+        verify(exactly = 1) { bannerText.text() }
+        verify(exactly = 2) { anyConstructed<RemoteViews>().setTextViewText(any(), primaryText()) }
+
+        notification.updateNotification(routeProgress)
+
+        verify(exactly = 2) { bannerText.text() }
+        verify(exactly = 2) { anyConstructed<RemoteViews>().setTextViewText(any(), primaryText()) }
+    }
+
+    @Test
+    fun whenUpdateNotificationCalledTwiceWithDifferentDataThenRemoteViewUpdatedTwice() {
+        val routeProgress = mockk<RouteProgress>(relaxed = true)
+        var primaryText = "Primary Text"
+        val primaryTextLambda = { primaryText }
+        val bannerText = mockBannerText(routeProgress, primaryTextLambda)
+        mockUpdateNotificationAndroidInteractions()
+
+        notification.updateNotification(routeProgress)
+
+        verify(exactly = 1) { bannerText.text() }
+        verify(exactly = 2) {
+            anyConstructed<RemoteViews>().setTextViewText(any(), primaryTextLambda())
+        }
```

Comment:
NIT: IMO, we don't need to duplicate verifications/assertions that are already performed in a different test